### PR TITLE
Practical enrolment

### PR DESCRIPTION
**Pull request by shaftoes**
added practical enrolment field to course records, database, and timetable parsing.

### CODE REVIEW

File: hs/Database/JsonParser.hs [hunk 1]
```diff
@@ -78,11 +80,19 @@ setTutEnrol course val =
    updateWhere [CoursesCode ==. course]
    [CoursesManualTutorialEnrolment =. Just val]
 
+<<<<<<< HEAD
```
Comment (Ian-Stewart-Binks) [inline hs/Database/JsonParser.hs]: <<<<<<

File: hs/WebParsing/TimeTableParser.hs [hunk 2]
```diff
@@ -206,13 +206,13 @@ processCourseTable course = do
...
+
+    setPracEnrol code (containsPrac sesh)
     mapM_ (insertLec session code) (lectures sesh)
     mapM_ (insertTut session code) (tutorials sesh)
   where
-    containsTut sesh =
-      foldl (\bool tut-> maybe False (T.isPrefixOf "T") (tutorialSection tut) || bool)
-            False (tutorials sesh)
+    containsTut sesh = foldl (\bool tut-> maybe False (T.isPrefixOf "T") (tutorialSection tut) || bool ) False (tutorials sesh)   
+    containsPrac sesh = foldl (\bool tut-> maybe False (T.isPrefixOf "P") (tutorialSection tut) || bool ) False (tutorials sesh)
```
Comment (Ian-Stewart-Binks) [inline hs/WebParsing/TimeTableParser.hs]: Is it possible to use `any` here?

File: hs/WebParsing/TimeTableParser.hs [hunk 3]
```diff
@@ -206,13 +206,13 @@ processCourseTable course = do
   runSqlite dbStr $ do
     runMigration migrateAll
     setTutEnrol code (containsTut sesh)
-    --setPracEnrol code (containsPractical sesh)
+
```
Comment (david-yz-liu) [inline hs/WebParsing/TimeTableParser.hs]: No need for a blank line here

File: hs/Database/Tables.hs [hunk 4]
```diff
@@ -158,12 +160,13 @@ instance FromJSON Course where
                <*> v .:  "name"
                <*> v .:? "exclusions"
                <*> v .:? "manualTutorialEnrolment"
+               <*> v .:? "manualPracticalEnrolment"
```
Comment (Ian-Stewart-Binks) [inline hs/Database/Tables.hs]: Are these really optional though?

**Pull request merged**